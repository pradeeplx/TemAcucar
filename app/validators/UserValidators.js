export default UserValidators = {
  first_name: {
    title: 'Nome',
    validate: [{
      validator: 'isLength',
      arguments: [1, 255],
    }]
  },
  last_name: {
    title: 'Sobrenome',
    validate: [{
      validator: 'isLength',
      arguments: [1, 255],
    }]
  },
  email: {
    title: 'E-mail',
    validate: [{
      validator: 'isLength',
      arguments: [1, 255],
    },
    {
      validator: 'isEmail',
      message: 'Complete seu e-mail',
    }]
  },
  password: {
    title: 'Senha',
    validate: [{
      validator: 'isLength',
      arguments: [8, 255],
      message: 'Use uma senha maior',
    }]
  },
  reset_password_token: {
    title: 'Código enviado',
    validate: [{
      validator: 'isLength',
      arguments: [1, 255],
    }]
  },
  address_thoroughfare: {
    title: 'Rua, Av., etc',
    validate: [{
      validator: 'isLength',
      arguments: [1, 255],
    }]
  },
  address_sub_thoroughfare: {
    title: 'Número',
    validate: [{
      validator: () => true,
    }]
  },
  address_complement: {
    title: 'Complemento',
    validate: [{
      validator: () => true,
    }]
  },
  address_sub_locality: {
    title: 'Complemento',
    validate: [{
      validator: () => true,
    }]
  },
  address_locality: {
    title: 'Cidade',
    validate: [{
      validator: 'isLength',
      arguments: [1, 255],
    }]
  },
  address_administrative_area: {
    title: 'Estado',
    validate: [{
      validator: 'isLength',
      arguments: [1, 255],
    }]
  },
  address_country: {
    title: 'País',
    validate: [{
      validator: 'isLength',
      arguments: [1, 255],
    }]
  },
  errorMessage: ({ id }) => {
    switch (id) {
      case 'not_found':
        return 'Usuário não encontrado.'
      case 'email_is_already_taken':
        return 'E-mail já cadastrado para outro usuário.'
      case 'unauthorized':
        return 'Código inválido. Confira seu e-mail ;)'
      default:
        return 'Oops! Ocorreu um erro ao acessar nosso servidor. Por favor, tente novamente.'
    }
  }
}
