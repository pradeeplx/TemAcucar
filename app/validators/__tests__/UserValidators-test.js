import 'react-native'
import React from 'react'
import UserValidators from '../UserValidators'

test('UserValidators object', () => {
  expect(UserValidators).toMatchSnapshot()
})

describe('UserValidators.errorMessage cases', () => {
  it('error.id is not_found', () => {
    expect(UserValidators.errorMessage({ id: 'not_found' })).toMatchSnapshot()
  })

  it('error.id is email_is_already_taken', () => {
    expect(UserValidators.errorMessage({ id: 'email_is_already_taken' })).toMatchSnapshot()
  })

  it('error.id is unauthorized', () => {
    expect(UserValidators.errorMessage({ id: 'unauthorized' })).toMatchSnapshot()
  })

  it('has a default errorMessage', () => {
    expect(UserValidators.errorMessage({ id: 'foo' })).toMatchSnapshot()
  })
})