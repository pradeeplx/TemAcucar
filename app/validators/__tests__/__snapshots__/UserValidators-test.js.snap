exports[`UserValidators.errorMessage cases error.id is email_is_already_taken 1`] = `"E-mail já cadastrado para outro usuário."`;

exports[`UserValidators.errorMessage cases error.id is not_found 1`] = `"Usuário não encontrado."`;

exports[`UserValidators.errorMessage cases error.id is unauthorized 1`] = `"Código inválido. Confira seu e-mail ;)"`;

exports[`UserValidators.errorMessage cases has a default errorMessage 1`] = `"Oops! Ocorreu um erro ao acessar nosso servidor. Por favor, tente novamente."`;

exports[`test UserValidators object 1`] = `
Object {
  "address_administrative_area": Object {
    "title": "Estado",
    "validate": Array [
      Object {
        "arguments": Array [
          1,
          255,
        ],
        "validator": "isLength",
      },
    ],
  },
  "address_complement": Object {
    "title": "Complemento",
    "validate": Array [
      Object {
        "validator": [Function],
      },
    ],
  },
  "address_country": Object {
    "title": "País",
    "validate": Array [
      Object {
        "arguments": Array [
          1,
          255,
        ],
        "validator": "isLength",
      },
    ],
  },
  "address_locality": Object {
    "title": "Cidade",
    "validate": Array [
      Object {
        "arguments": Array [
          1,
          255,
        ],
        "validator": "isLength",
      },
    ],
  },
  "address_sub_locality": Object {
    "title": "Complemento",
    "validate": Array [
      Object {
        "validator": [Function],
      },
    ],
  },
  "address_sub_thoroughfare": Object {
    "title": "Número",
    "validate": Array [
      Object {
        "validator": [Function],
      },
    ],
  },
  "address_thoroughfare": Object {
    "title": "Rua, Av., etc",
    "validate": Array [
      Object {
        "arguments": Array [
          1,
          255,
        ],
        "validator": "isLength",
      },
    ],
  },
  "email": Object {
    "title": "E-mail",
    "validate": Array [
      Object {
        "arguments": Array [
          1,
          255,
        ],
        "validator": "isLength",
      },
      Object {
        "message": "Complete seu e-mail",
        "validator": "isEmail",
      },
    ],
  },
  "errorMessage": [Function],
  "first_name": Object {
    "title": "Nome",
    "validate": Array [
      Object {
        "arguments": Array [
          1,
          255,
        ],
        "validator": "isLength",
      },
    ],
  },
  "last_name": Object {
    "title": "Sobrenome",
    "validate": Array [
      Object {
        "arguments": Array [
          1,
          255,
        ],
        "validator": "isLength",
      },
    ],
  },
  "password": Object {
    "title": "Senha",
    "validate": Array [
      Object {
        "arguments": Array [
          8,
          255,
        ],
        "message": "Use uma senha maior",
        "validator": "isLength",
      },
    ],
  },
  "reset_password_token": Object {
    "title": "Código enviado",
    "validate": Array [
      Object {
        "arguments": Array [
          1,
          255,
        ],
        "validator": "isLength",
      },
    ],
  },
}
`;
